components/graphs/ThresholdGraph.tsx

```diff
@@ -32,13 +35,38 @@ const ThresholdGraph: FC<{ siteUUID: string }> = ({ siteUUID }) => {
     useSiteData(siteUUID);
   const { currentTime } = useTime(latitude, longitude);
 
+  const [graphData, setGraphData] = useState(
+    forecastData
+      ? forecastDataOverDateRange(
+          forecastData.forecast_values,
+          getGraphStartDate(currentTime),
+          getGraphEndDate(currentTime)
+        )
+      : []
+  );
+
+  useEffect(() => {
+    if (graphData.length === 0) {
+      console.log('updating');
+      setGraphData(
+        forecastData
+          ? forecastDataOverDateRange(
+              forecastData.forecast_values,
+              getGraphStartDate(currentTime),
+              getGraphEndDate(currentTime)
+            )
+          : []
+      );
+    }
+  }, [forecastData, currentTime, graphData]);
```

Comment:
Instead of a `useEffect` here and a `useState` above, can `graphData` just be a `const` defined to be equal to
```
forecastData
          ? forecastDataOverDateRange(
              forecastData.forecast_values,
              getGraphStartDate(currentTime),
              getGraphEndDate(currentTime)
            )
          : []
```

or does that not work?

Reply:
So until we did this, the ThresholdGraph labels (current time marker, threshold label) weren't showing up. @ashayp22 helped us figure out that the were likely not rendering because at first, graphData did not contain any data until forecastData was read with useSiteData so there was nothing to label, and then the problem (probably) was is that they didn't update when graphData updated.

Reply:
Thanks for the info, yeah this works then